### ISSUE TITLE
sbe-tool nuget package is DOA

### ISSUE THREAD

**Issue by damageboy** [real-logic/simple-binary-encoding#698]
When using the sbe-tool from the latest [sbe-tool nuget package](https://www.nuget.org/packages/sbe-tool/0.1.12.1-beta-1): 0.1.12-1-beta-1, code generation immediately fails

```java
Exception in thread "main" java.lang.NoClassDefFoundError: org/agrona/MutableDirectBuffer
        at uk.co.real_logic.sbe.SbeTool.<clinit>(SbeTool.java:82)
Caused by: java.lang.ClassNotFoundException: org.agrona.MutableDirectBuffer
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 1 more
```

It seems that the nuget package contains the WRONG jar, it should have contained the jar from the: `sbe-all/build/libs/` folder which contains all the required java deps, but instead it uses the jar from the `sbe-tool/build/libs/` folder which is partial as far as running the complete sbe-tool is...

Would it be OK for me submit a PR that fixes this and for someone who has access to the nuget account (I presume @billsegall) to publish an updated nuget package?

As far as I can tell, this thing is dead out of water from nuget....

**Comment by mjpt777** [comment]
It looks like their are a number of issues worth addressing with the C# implementation. It would be good for those interested to jointly agree on how best to improve it and I'll happily support that.

**Comment by damageboy** [comment]
I wouldn't mind taking a few concentrated days to try and bring it more up to speed..., looks to me like we need:

* CI support for the C# parts
* Automatic packaging / myget feed per build
* Running unit tests on the build server (e.g. building + running everything on Linux as part of the current travis build)

Would you like me to first try and get the C# parts of the project into a working state?

**Comment by mjpt777** [comment]
That would be great. Then maybe @billsegall, @JPWatson, @ZachBray and whoever else is interested can feedback and agree changes.

**Comment by damageboy** [comment]
OK, the only annoyance is that I've already done some build related touch-ups as part of https://github.com/real-logic/simple-binary-encoding/pull/695

It would be easier to me if someone could review that and merge that so I could branch off of that to start the CI related stuff, but if that's not possible in the short-term, let me know and I'll redo that work in a new branch dedicated to the CI status

**Comment by billsegall** [comment]
It would seem I broke this with https://github.com/real-logic/simple-binary-encoding/commit/8b9bda8870321c14e04490983b8b09b4348fa6ad

Can anyone provide context on why the '-all' jar stopped building. I had wrongly assumed it had just changed name.

I'm on the road holidaying with limited internet/facilities for the next three weeks but will try to chime in when I can.

**Comment by damageboy** [comment]
@billsegall I think I have something working in https://github.com/real-logic/simple-binary-encoding/pull/699

There's a ton of small fixes there, and once travis does its thing, I think you can push a new nuget from there...

**Comment by damageboy** [comment]
travis now seems to build this properly... (https://github.com/real-logic/simple-binary-encoding/pull/699)

Once merged I think we can release a new nuget package after someone can test it except for me :)

**Comment by mjpt777** [comment]
What is the status of the nuget package? If we do a new release in a few days what is required?

**Comment by billsegall** [comment]
I'll be back at my desk on Monday and will publish a new version. Long term
we need to make this possible at a project level.

Bill

**Comment by mjpt777** [comment]
Thanks @billsegall. Should we hold off on a release until after Monday?

**Comment by billsegall** [comment]
Go for it now. If there is a problem we can fix it after and it's nice to
be close to an official release.

Bill.

**Comment by damageboy** [comment]
Great!
All I can say is that I've been using it from an internal feed, and as far as I can tell it's working well (codegen + generated code).

**Comment by billsegall** [comment]
Heh. Won't be Monday, I'm having one of those air travel nightmares on a
super long haul flight.

Bill

**Comment by mjpt777** [comment]
Sorry to hear that. Travel can be a pain at times. Hopefully gets better.

**Comment by damageboy** [comment]
Other than the official package at nuget.org, I can also setup some sort of a ci nuget feed on myget or something of that source, if that is of any interest...

I would make consuming ci packages a real possibility, although that might be over the top.

**Comment by billsegall** [comment]
The sbe-dll package is uploaded and is being verified. I've created a PR
with the minor changes for the CSharp release.

I notice that package name has changed from sbe-tool to sbe-dll and I'm
wondering how this will affect users of the nuget package and automatic
updates?

I'm in two minds about myget packages. I think it would be very useful to
be able to have the travid process build, upload, and test against the
uploaded package but I worry people might use it and think it stable.

Bill.

**Comment by damageboy** [comment]
@billsegall I was pretty sure I did not change anything in the nuget spec, and I really tried to meticulously copy the nuspec to the appropriate .csproj properties during the "porting" process.

Let me double check and get back to you.

**Comment by billsegall** [comment]
I think it's still open and will have a dig next week. I think I've decided
that referring the name is a good idea.

Bill.

**Comment by billsegall** [comment]
Martin,

I found some time, and with the now merged #707
<https://github.com/real-logic/simple-binary-encoding/pull/707> and newly
published sbe-tool 1.13.1 <https://www.nuget.org/packages/sbe-tool/>, I
think this can now be closed.
